quickstarts/microsoft.network/application-gateway-v2-autoscale-create/main.bicep

```diff
@@ -0,0 +1,172 @@
+@description('Virtual Network name')
+param virtualNetworkName string = 'Application-Vnet'
+
+@description('Virtual Network address range')
+param vnetAddressPrefix string = '10.0.0.0/16'
+
+@description('Name of the subnet')
+param subnetName string = 'ApplicationGatewaySubnet'
+
+@description('Subnet address range')
+param subnetPrefix string = '10.0.0.0/24'
+
+@description('Application Gateway name')
+param applicationGatewayName string = 'applicationGatewayV2'
+
+@description('Minimum instance count for Application Gateway')
+param minCapacity int = 2
+
+@description('Maximum instance count for Application Gateway')
+param maxCapacity int = 10
+
+@description('Application Gateway Frontend port')
+param frontendPort int = 80
+
+@description('Application gateway Backend port')
+param backendPort int = 80
+
+@description('Back end pool ip addresses')
+param backendIPAddresses array = [
+  {
+    IpAddress: '10.0.0.4'
+  }
+  {
+    IpAddress: '10.0.0.5'
+  }
+]
+
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+@description('Cookie based affinity')
+param cookieBasedAffinity string = 'Disabled'
+
+@description('Location for all resources.')
+param location string = resourceGroup().location
+
+var appGwPublicIpName = '${applicationGatewayName}-pip'
+var appGwSize = 'Standard_v2'
+
+resource virtualNetwork 'Microsoft.Network/virtualNetworks@2020-06-01' = {
+  name: virtualNetworkName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        vnetAddressPrefix
+      ]
+    }
+    subnets: [
+      {
+        name: subnetName
+        properties: {
+          addressPrefix: subnetPrefix
+        }
+      }
+    ]
+  }
+}
+
+resource publicIP 'Microsoft.Network/publicIPAddresses@2020-06-01' = {
+  name: appGwPublicIpName
+  location: location
+  sku: {
+    name: 'Standard'
+  }
+  properties: {
+    publicIPAllocationMethod: 'Static'
+  }
+}
+
+resource applicationGateway 'Microsoft.Network/applicationGateways@2020-06-01' = {
+  name: applicationGatewayName
+  location: location
+  properties: {
+    sku: {
+      name: appGwSize
```

Comment:
are both of these required?  they are set to the same value but one is hard-coded, and I suspect if the param changes this will break...

Reply:
You tell me.  The docs don't say much:
![image](https://user-images.githubusercontent.com/6913354/122134977-83177900-cdf4-11eb-9c9c-21e63609a306.png)
https://docs.microsoft.com/en-us/azure/templates/microsoft.network/applicationgateways?tabs=json

This is straight from the original quickstart's json.

Reply:
![image](https://user-images.githubusercontent.com/6913354/122135045-a6422880-cdf4-11eb-8169-1ed7e60203fe.png)